lib/models/project.js

```diff
@@ -408,16 +438,83 @@ class Project {
   }
 
   /**
-    Discovers all addons for this project and stores their names and
-    package.json contents in this.addonPackages as key-value pairs
-
-    @private
-    @method discoverAddons
+   * Discovers all addons for this project and stores their names and
+   * package.json contents in this.addonPackages as key-value pairs.
+   *
+   * Any packageInfos that we find that are marked as not valid are excluded.
+   *
+   * @private
+   * @method discoverAddons
    */
   discoverAddons() {
-    let addonsList = this.addonDiscovery.discoverProjectAddons(this);
+    let pkgInfo = this._packageInfoCache.getEntry(this.root);
+
+    let addonPackageList = [];
+
+    if (this.isEmberCLIAddon()) {
+      addonPackageList.push(pkgInfo);
+    }
+
+    let cliInRepoAddons = pkgInfo.cliEntry
+      ? pkgInfo.cliEntry.inRepoAddons
+      : null;
+    if (cliInRepoAddons) {
+      cliInRepoAddons.forEach(addon => addonPackageList.push(addon));
+    }
+
+    if (pkgInfo.internalAddons) {
+      pkgInfo.internalAddons.forEach(addon => addonPackageList.push(addon));
+    }
+
+    // The 'dependencyPackages' object contains ALL dependencies, not just those
+    // that are addons. We want only the addons
+    if (pkgInfo.dependencyPackages) {
+      Object.keys(pkgInfo.dependencyPackages).forEach(name => {
```

Comment:
could probably use `filter`

Reply:
Ya, until our lowest node LTS support version is 8. I think we are stuck without.

I don't think we need to add a polyfil, what is written seems fine.